umpleonline/ump/GeometricSystem.ump

```diff
@@ -115,143 +115,154 @@ class NonPolyhedra{
 class Circle{
 	isA CurvedShape;
 	double radius;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(CurvedShape object){/*implementation */}
-	boolean isEqual(CurvedShape object){/*implementation */}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(CurvedShape object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
 }
 class Ellipse {
 	isA CurvedShape;
 	double majorAxis;
 	double minorAxis;
 	Point fFocus;
 	Point gFocus;
-	Point[2] getFoci(){/*implementation */}
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	Line getTangent(){/*implementation */}
-	boolean isLessThan(CurvedShape object){}
-	boolean isEqual(CurvedShape object){}
+	Point[] getFoci(){return new Point[2];}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	Line getTangent(){return new Line(fFocus, gFocus);}
+	boolean isLessThan(CurvedShape object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
 }
 class Quadrilateral{
 	isA Polygon;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(Polygon object){}
-	boolean isEqual(Polygon object){}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(Polygon object){return true;}
+	boolean isEqual(Polygon object){return true;}
+	boolean isEqual(CurvedShape object){return false;}
```

Comment:
@vahdat-ab Added the @@@skipjavacompile directive to GeometricSystem.ump. This can be removed when the issue with TComparable is resolved.